pkg/pkgupdates/pkgupdates.jsx

```diff
@@ -0,0 +1,558 @@
+                              this.setState({state: "updateError"});
+                          });
+            })
+            // this Should Not Fail™, so don"t bother about the slightly mislabeled state here
+            .fail(this.handleLoadError);
+    }
+
+    renderContent() {
+        switch (this.state.state) {
+            case "loading":
+            case "locked":
+                if (this.state.loadPercent)
+                    return (
+                        <div className="progress">
+                          <div className="progress-bar" role="progressbar"
+                               style={{width: this.state.loadPercent + "%"}}>
+                          </div>
+                      </div>)
+                else
+                    return <div className="spinner spinner-lg" />;
+
+            case "available":
+                return (
+                    <div>
+                        <table width="100%">
+                            <tr>
+                                <td><h2>{_("Available Packages")}</h2></td>
+                                <td className="text-right">
+                                    { this.state.haveSecurity
+                                      ? <button className="btn btn-default"
+                                                 onClick={() => this.applyUpdates(true)}>
+                                            {_("Install security updates")}
+                                         </button>
+                                      : null
+                                    }
+                                    &nbsp; &nbsp;
+                                    <button className="btn btn-primary"
+                                            onClick={() => this.applyUpdates(false)}>
+                                        {_("Install all updates")}
+                                    </button>
+                                </td>
+                            </tr>
+                        </table>
+                        { this.state.cockpitUpdate
+                          ? <div className="alert alert-warning">
+                                <span className="pficon pficon-warning-triangle-o"></span>
+                                <span>
+                                    <strong>{_("Cockpit itself will be updated.")}</strong>
+                                    &nbsp;
+                                    {_("When you get disconnected, the updates will continue in the background. You can reconnect and resume watching the update progress.")}
+                                </span>
+                            </div>
+                          : null
+                        }
+                        <UpdatesList updates={this.state.updates} />
+                    </div>
+                );
+
+            case "loadError":
+            case "updateError":
+                return this.state.errorMessages.map(m => <pre>{m}</pre>);
```

Comment:
I'm not sure - if the upgrade fails, you should really investigate on the command line, as there is no interactivity in the PackageKit upgrade. The user might try to re-run the update, but that might make things worse (or at least not better). I'm okay with adding some "Acknowledge" or "Go back" button there in case a retry does help. @garrett, do you have some input on that?

I added it as a TODO item for now, but I don't think this necessarily needs to block this PR.